src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_TabKey.cs

```diff
@@ -67,60 +64,91 @@ private bool TryInvokeSnippetCompletion(TabKeyCommandArgs args, CompletionServic
 
             var text = subjectBuffer.AsTextContainer().CurrentText;
 
-            // Delete the ? and invoke completion
+            // If the user types "<line start><spaces><question><tab>"
```

Comment:
> QuestionMarkIsPrecededByIdentifierAndWhitespace seems weirdly complicated for this task, unless I'm missing some nuance

I was just trying to preserve semantics.  If this code really is: "i just need to pop up if i'm after ```id?<tab>``` then i'm happy to simplify.